pkg/networkserver/grpc_deviceregistry.go

```diff
@@ -186,6 +223,28 @@ func (ns *NetworkServer) Get(ctx context.Context, req *ttnpb.GetEndDeviceRequest
 		dev.Session.SessionKeys = sk
 	}
 
+	if dev.PendingMACState.GetQueuedJoinAccept() != nil && ttnpb.HasAnyField(req.FieldMask.Paths,
+		"pending_mac_state.queued_join_accept.keys.app_s_key.key",
+		"pending_mac_state.queued_join_accept.keys.f_nwk_s_int_key.key",
+		"pending_mac_state.queued_join_accept.keys.nwk_s_enc_key.key",
+		"pending_mac_state.queued_join_accept.keys.s_nwk_s_int_key.key",
+	) {
+		appSKey := dev.PendingMACState.QueuedJoinAccept.Keys.AppSKey
+		dev.PendingMACState.QueuedJoinAccept.Keys.AppSKey = nil
+		sk, err := cryptoutil.UnwrapSelectedSessionKeys(ctx, ns.KeyVault, dev.PendingMACState.QueuedJoinAccept.Keys, "pending_mac_state.queued_join_accept.keys", req.FieldMask.Paths...)
+		if err != nil {
+			return nil, err
+		}
+		if ttnpb.HasAnyField(req.FieldMask.Paths, "pending_mac_state.queued_join_accept.keys.app_s_key.key") && appSKey != nil {
+			key, err := cryptoutil.UnwrapAES128Key(ctx, *appSKey, ns.KeyVault)
+			if err != nil && !errors.IsNotFound(err) {
+				return nil, err
+			} else if err == nil {
+				sk.AppSKey = &ttnpb.KeyEnvelope{Key: &key}
+			}
+		}
```

Comment:
Can you enlighten me on what NS is doing with AppSKey here? Regardless of pending or not, NS should not be able to unwrap the AppSKey